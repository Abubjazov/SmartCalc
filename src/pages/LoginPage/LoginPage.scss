@import '../../variables';

.login-page {
	background: linear-gradient(
		180deg,
		rgba(254, 254, 254, 1) 0%,
		rgba(238, 238, 238, 1) 100%
	);

	.container {
		padding: 11vh 0 1vh 0;
		min-height: 100vh;
		max-width: calc(100vw - 2vw);
		display: grid;
		grid-template-columns: 1fr;
		grid-template-rows: 1fr;
		justify-content: center;
		align-items: center;

		@media (max-width: 900px) {
			padding: 6vh 0 0 0;
		}

		svg {
			justify-self: center;
		}

		a {
			border-radius: 20px;
			padding-top: 11px;
			width: 100px;
			height: 40px;
			text-align: center;
			font-size: 18px;
			line-height: 18px;
			background-color: $main-color;
			color: $white;
			text-decoration: none;
			transition: all 0.1s linear;
		}
	}
}
