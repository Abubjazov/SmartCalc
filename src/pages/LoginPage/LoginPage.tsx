import { Helmet } from 'react-helmet'

import { useTypedSelector } from '../../hooks/useTypedSelector'
import { Login, Spinner } from '../../components'

import './LoginPage.scss'

export const LoginPage = (): JSX.Element => {
	const { status } = useTypedSelector(state => state.auth)

	return (
		<main className='login-page'>
			<Helmet>
				<title>Авторизация</title>
			</Helmet>

			<div className='container'>
				{status === 'loading' ? <Spinner /> : null}
				{status === 'waiting' || status === 'error' ? <Login /> : null}
			</div>
		</main>
	)
}
