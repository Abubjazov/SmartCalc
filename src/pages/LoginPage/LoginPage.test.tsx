import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import { BrowserRouter } from 'react-router-dom'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'

import { LoginPage } from './LoginPage'

const middlewares = [thunk]
const mockStore = configureStore(middlewares)

describe('Component: LoginPage', () => {
	test('should render LoginPage default', () => {
		const store = mockStore({
			auth: {
				status: 'waiting',
				error: null,
			},
		})

		const { asFragment } = render(
			<Provider store={store}>
				<BrowserRouter>
					<LoginPage />
				</BrowserRouter>
			</Provider>
		)

		expect(asFragment()).toMatchSnapshot()
	})

	test('should render LoginPage with loading spinner', () => {
		const store = mockStore({
			auth: {
				status: 'loading',
				error: null,
			},
		})

		const { asFragment } = render(
			<Provider store={store}>
				<BrowserRouter>
					<LoginPage />
				</BrowserRouter>
			</Provider>
		)

		expect(asFragment()).toMatchSnapshot()
	})

	test('should render LoginPage with error message', () => {
		const store = mockStore({
			auth: {
				status: 'error',
				error: 'ERROR*Test Error String>>test',
			},
		})

		const { asFragment } = render(
			<Provider store={store}>
				<BrowserRouter>
					<LoginPage />
				</BrowserRouter>
			</Provider>
		)

		expect(asFragment()).toMatchSnapshot()
	})
})
