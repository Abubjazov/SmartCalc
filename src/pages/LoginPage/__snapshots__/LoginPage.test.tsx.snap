// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: LoginPage should render LoginPage default 1`] = `
<DocumentFragment>
  <main
    class="login-page"
  >
    <div
      class="container"
    >
      <div
        class="login-widget"
      >
        <input
          class="null"
          placeholder="your@email.com"
          type="email"
          value=""
        />
        <input
          class="null"
          placeholder="password"
          type="password"
          value=""
        />
        <button
          disabled=""
        >
          Войти
        </button>
      </div>
    </div>
  </main>
</DocumentFragment>
`;

exports[`Component: LoginPage should render LoginPage with error message 1`] = `
<DocumentFragment>
  <main
    class="login-page"
  >
    <div
      class="container"
    >
      <main>
        <div
          class="error-message"
          role="alert"
        >
          <p
            class="error-message-p"
            tabindex="0"
          >
            ERROR
          </p>
          <p
            class="error-message-p"
            tabindex="0"
          >
            Test Error String
          </p>
          <a
            class="error-message-a"
            href="/smartcalc/"
          >
            Вернуться на главную
          </a>
        </div>
      </main>
    </div>
  </main>
</DocumentFragment>
`;

exports[`Component: LoginPage should render LoginPage with loading spinner 1`] = `
<DocumentFragment>
  <main
    class="login-page"
  >
    <div
      class="container"
    >
      <svg
        class="spinner"
        height="200"
        preserveAspectRatio="xMidYMid"
        viewBox="0 0 100 100"
        width="200"
      >
        <path
          d="M24.3 30C11.4 30 5 43.3 5 50s6.4 20 19.3 20c19.3 0 32.1-40 51.4-40 C88.6 30 95 43.3 95 50s-6.4 20-19.3 20C56.4 70 43.6 30 24.3 30z"
          fill="none"
          stroke="#cccccc"
          stroke-dasharray="42.76482137044271 42.76482137044271"
          stroke-linecap="round"
          stroke-width="8"
        >
          <animate
            attributeName="stroke-dashoffset"
            dur="1s"
            keyTimes="0;1"
            repeatCount="indefinite"
            values="0;256.58892822265625"
          />
        </path>
      </svg>
    </div>
  </main>
</DocumentFragment>
`;
